### PR TITLE
Fixed array_merge_recursive issue in Phalcon\Config\Adapter\Ini.zep

### PR DESCRIPTION
**Pull request by hinablue**
- [x] Make sure that the PR passes in Travis CI to make the process more efficient.
- [x] Only use tabs for indentation.
- [x] if needed, rebase to the proper branch before submitting your pull request.
      If it doesn't merge cleanly with master you may be asked to rebase your changes.
- [x] Don't put submodule updates in your pull request unless they are to landed commits.
- [ ] Add tests relevant to the fixed bug or new feature. See our testing guide for
      more information.
- [x] Phalcon 2 is written in Zephir, please do not submit commits that modify C generated
      files directly or those whose functionality/fixes are implemented in the C
      programming language
- [x] Remove any change to ext/kernel / *.zep.c / *.zep.h files before submitting the PR

There (maybe) is the PHP core issue, not Phalcon. And I rewrite the array_merge_recursive like here,
[http://php.net/manual/en/function.array-merge-recursive.php#96201](http://php.net/manual/en/function.array-merge-recursive.php#96201)

Use `Phalcon\Config\Adapter\Ini`, If I have ini file like,

```
[step]
step1.1 = A
step1.2 = B
step1.3 = C
```

Output will be,

```
array(1) {
  ["step"]=>
  array(1) {
    ["step1"]=>
    array(2) {
      [1]=>
      string(1) "B"
      [2]=>
      string(1) "C"
    }
  }
}
```

The 'A' will missing, but If,

```
[step]
step1.5 = X
step1.1 = A
step1.2 = B
step1.3 = C
```

Output will be,

```
array(1) {
  ["step"]=>
  array(1) {
    ["step1"]=>
    array(2) {
      [5]=>
      string(1) "X"
      [1]=>
      string(1) "A"
      [2]=>
      string(1) "B"
      [3]=>
      string(1) "C"
    }
  }
}
```

This issue is PHP's array_merge_recursive (maybe), so I rewrite an new function to do this in `ini.zep` and output correct what we want.

There is ini file,

```
[qq]
step1.1 = A
step1.2 = B
step1.3 = G
step2.1 = A
step2.2 = B
step2.3 = C
step12[3] = AA
step12[4] = BB
step12[5] = CC
step5.3 = AAA
step5.4 = BBBB
step5.1 = GGGG
step5.2 = TTTT
```

And output will be,

```
object(Phalcon\Config\Adapter\Ini)#1 (1) {
  ["qq"]=>
  object(Phalcon\Config)#2 (4) {
    ["step1"]=>
    object(Phalcon\Config)#3 (3) {
      ["1"]=>
      string(1) "A"
      ["2"]=>
      string(1) "B"
      ["3"]=>
      string(1) "G"
    }
    ["step2"]=>
    object(Phalcon\Config)#4 (3) {
      ["1"]=>
      string(1) "A"
      ["2"]=>
      string(1) "B"
      ["3"]=>
      string(1) "C"
    }
    ["step12"]=>
    object(Phalcon\Config)#5 (3) {
      ["3"]=>
      string(2) "AA"
      ["4"]=>
      string(2) "BB"
      ["5"]=>
      string(2) "CC"
    }
    ["step5"]=>
    object(Phalcon\Config)#6 (4) {
      ["3"]=>
      string(3) "AAA"
      ["4"]=>
      string(4) "BBBB"
      ["1"]=>
      string(4) "GGGG"
      ["2"]=>
      string(4) "TTTT"
    }
  }
}
```

Thanks

### CODE REVIEW

File: phalcon/config/adapter/ini.zep [hunk 1]
```diff
@@ -96,7 +96,7 @@ class Ini extends Config
 					let sections[] = this->_parseIniString((string)path, lastValue);
 				}
 				if count(sections) {
-					let config[section] = call_user_func_array("array_merge_recursive", sections);
+					let config[section] = call_user_func_array([this, "_array_merge_recursive_distinct"], sections);
```
Comment (Jurigag) [inline phalcon/config/adapter/ini.zep]: Just normally call a method...

**Review by Jurigag — CHANGES_REQUESTED** [review]

File: phalcon/config/adapter/ini.zep (hunk 1 above)
Comment (sergeyklay) [inline phalcon/config/adapter/ini.zep]: As you can see the tests failed

**Comment by hinablue** [comment]
Original PHP `parse_ini_file` when use **INI_SCANNER_NORMAL** WILL NOT REPLACE variable in ini file if defined, but **INI_SCANNER_RAW** will be.

And `Phalcon\Config\Adapter\Ini` when use **INI_SCANNER_NORMAL** will always replace variable in ini if defined.

php file,
```
define('TEST_CONST', 'foo');
```

ini file,
```
[section]
test = TEST_CONST"hello"
```

1. `parse_ini_file` with **INI_SCANNER_NORMAL** will output:
```
array(1) {
  ["section"]=>
  array(1) {
    ["test"]=>
    string(17) "TEST_CONST"hello""
  }
}
```

2. `parse_ini_file` with **INI_SCANNER_RAW** will output:
```
array(1) {
  ["section"]=>
  array(1) {
    ["test"]=>
    string(8) "foohello"
  }
}
```

3. `Phalcon\Config\Adapter\Ini` with **INI_SCANNER_NORMAL** will output:
```
object(Phalcon\Config\Adapter\Ini)#1 (1) {
  ["section"]=>
  object(Phalcon\Config)#2 (1) {
    ["test"]=>
    string(8) "foohello"
  }
}
```

So, whatever **INI_SCANNER_RAW** or **INI_SCANNER_NORMAL** with the `Phalcon\Config\Adapter\Ini` it always output like RAW's do?

If yes, I will fixed and fix this PR :-)

**Comment by Jurigag** [comment]
So this can theoretically break BC ?

**Pull request closed without merging**